numpy/lib/function_base.py

```diff
@@ -3635,7 +3869,19 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
     else:
         zerod = False
 
-    # prepare a for partitioning
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.size < 10:
+        for i in range(q.size):
+            if q[i] < 0. or q[i] > 1.:
+                raise ValueError("Quantiles must be in the range [0,1], "
+                                 "or percentiles must be [0,100].")
+    else:
+        # faster than any()
+        if np.count_nonzero(q < 0.) or np.count_nonzero(q > 1.):
+            raise ValueError("Quantiles must be in the range [0,1] "
+                             "or percentiles must be [0,100].")
```

Comment:
This looks like a incorrect rebase - we already check this at the top level, I think?

Reply:
You're right, we already checked this at the top level.  I can remove it.